include/rocksdb/db.h

```diff
@@ -131,6 +131,26 @@ struct IngestExternalFileArg {
   std::vector<std::string> files_checksum_func_names;
   // A hint as to the temperature for *reading* the files to be ingested.
   Temperature file_temperature = Temperature::kUnknown;
+  // EXPERIMENTAL: When specified, existing keys in the given range will be
+  // cleared atomically as part of the ingestion, where the ingested files are
+  // logically applied on top of the cleared key range.
+  // * When a range is specified, all the external files in this batch must
+  //   be contained in that key range.
+  // * Checks for memtable overlap and possible blocking flush will apply
+  //   to this range (not just the file ranges).
+  // * Not compatible with ingest_behind=true.
+  // * When options.snapshot_consistency = false, the range is cleared
+  // similarly to DeleteFilesInRange, but fails if any files overlap the range
+  // only partially.
+  //   * It is recommended to use fail_if_not_bottommost_level=true to ensure
+  //     data in the key range is ingested to a single compacted level (the
+  //     last level). (fail_if_not_bottommost_level=false allows overlap between
+  //     the ingested files.)
+  // * options.snapshot_consistency = true is not yet supported.
+  // BUG: the upper bound of the range may be interpreted as inclusive or
+  // exclusive, so it is best not to depend on one or the other until it is
+  // sorted out.
+  std::optional<Range> atomic_replace_range;
```

Comment:
How do you specify the entire key range? Do a SeekToLast() and increment the key?

Reply:
You'd need to increment the greater of largest existing key and largest key being ingested. It's not pretty, but the description tells why I think paralleling DeleteRange is the best long term approach. We should fix DeleteRange (and atomic_replace_range in the same way) to make extreme bounds easier.

Reply:
Hmm I guess I could go ahead and make atomic_replace_range a RangePtr and support special combination of snapshot_consistency=false and both bounds nullptr. That shouldn't be too difficult.

Reply:
That sounds good. We have a similar thing in `CompactRange` where `begin` and `end` both set to nullptr means the entire range, so its not without precedent.